keps/sig-node/4010-head-request-probes/README.md

```diff
@@ -0,0 +1,265 @@
+
+- Documentation: Clear documentation will be provided that outlines the supported Kubernetes versions for the new feature. This documentation will include information on compatibility between different Kubernetes components.
+
+- Compatibility Checks: During the upgrade process, compatibility checks will be performed to ensure that the Kubernetes cluster components are running compatible versions. These checks will help identify any version skew issues and provide guidance on resolving them before enabling the new feature.
+
+## Production Readiness Review Questionnaire
+
+Before enabling support for HTTP HEAD requests in readiness and liveness probes in a production environment, the following aspects will be assessed through a readiness review questionnaire:
+
+### Feature Enablement and Rollback
+
+- What flags or configuration settings are required to enable the new feature?
+
+A feature flag `HeadRequestProbes` will be introducded.
+
+- Can the feature be disabled or rolled back if issues arise?
+
+Yes.
+
+- Are there any specific considerations for enabling or disabling the feature on a live cluster?
+
+No.
+
+### Rollout, Upgrade, and Rollback Planning
+
+- What is the recommended rollout strategy for enabling the new feature?
+
+Update to the version which has the new feature implemented and enable the feature gate.
+
+- Are there any specific considerations or steps to follow during the rollout process?
+
+No.
+
+- Are there any backward compatibility concerns to address during the rollout?
+
+No.
+
+### Monitoring Requirements
+
+- Are there any additional monitoring requirements specific to HTTP HEAD-based probes?
+
+Metrics need to be checked. If GET vs. HEAD differ, new metrics must be implemented.
+
+### Dependencies
+
+- Are there any dependencies on other Kubernetes features, components, or external systems?
+
+No.
+
+- Are there any specific version requirements for these dependencies?
+
+No.
+
+### Scalability
+
+- How does the new feature impact the scalability of Kubernetes clusters?
+
+It does not impact the scalability.
+
+- Are there any scalability considerations or limitations to be aware of?
+
+No.
+
+### Troubleshooting
+
+- What are the common issues or failure scenarios related to the new feature?
+
+That the HEAD endpoint is not implemented at the application (obvious).
+
+- What troubleshooting steps can be taken to diagnose and resolve these issues?
+
+Request the health endpoint with a HEAD request directly.
+
+- Are there any recommended logs or diagnostic information that should be collected for troubleshooting purposes?
+
+No.
+
+## Implementation History
+
+- 16-05-2023: Proposal drafted.
+
+## Drawbacks
+
+The introduction of support for HTTP HEAD requests in readiness and liveness probes may have the following drawbacks:
+
+- Increased Complexity: The new feature may add complexity to the Kubernetes codebase, potentially affecting maintenance and troubleshooting efforts.
+
+## Alternatives
```

Comment:
one alternative is to ask app developer to implement `GET` endpoint that doesn't have a body. Functionally it will be the same, would it? Please describe why having http head built-in is a better solution.

Reply:
Jeremias described it in https://github.com/kubernetes/enhancements/pull/4012#discussion_r1222067773